File: webapp/src/datasource/models/Booking.ts
import BaseModel from './BaseModel'

export default class Booking extends BaseModel {
    parentName: string
    parentEmail: string
    parentPhone: string
    studentName: string
    studentAge: string
    isContacted: boolean
    notes: string
    constructor(
        id: string,
        parentName: string,
        parentEmail: string,
        parentPhone: string,
        studentName: string,
        studentAge: string,
        isContacted: boolean,
        notes: string
    ) {
        super(id)
        this.parentName = parentName
        this.parentEmail = parentEmail
        this.parentPhone = parentPhone
        this.studentName = studentName
        this.studentAge = studentAge
        this.isContacted = isContacted
        this.notes = notes
    }
}

File: java/backend/src/main/java/com/thelittlebigtalents/backend/model/impl/Booking.java
/* (C)2024 */
package com.thelittlebigtalents.backend.model.impl;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.thelittlebigtalents.backend.model.api.AbstractPersistableDocument;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class Booking extends AbstractPersistableDocument {
    private String parentName;
    private String parentEmail;
    private String parentPhone;
    private String studentName;
    private int studentAge;

    @JsonProperty(value = "isContacted")
    private boolean isContacted;

    private String notes;
}
